<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OpenGLOptionsDialog</class>
 <widget class="QDialog" name="OpenGLOptionsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>320</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGL 3.0 renderer options</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Render behavior</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2" columnstretch="3,1">
      <item row="1" column="0">
       <widget class="QRadioButton" name="syncToFramerate">
        <property name="text">
         <string>Use target framerate:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="targetFps">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="suffix">
         <string> fps</string>
        </property>
        <property name="minimum">
         <number>15</number>
        </property>
        <property name="maximum">
         <number>240</number>
        </property>
        <property name="value">
         <number>60</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="vsync">
        <property name="text">
         <string>VSync</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QRadioButton" name="syncWithVideo">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Render each frame immediately, in sync with the emulated display.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;This is the recommended option if the shaders in use don't utilize frametime for animated effects.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Synchronize with video</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QSlider" name="fpsSlider">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="minimum">
         <number>15</number>
        </property>
        <property name="maximum">
         <number>240</number>
        </property>
        <property name="value">
         <number>60</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="invertedAppearance">
         <bool>false</bool>
        </property>
        <property name="tickPosition">
         <enum>QSlider::NoTicks</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Shaders</string>
     </property>
     <layout class="QGridLayout" name="gridLayout" columnstretch="3,1">
      <item row="2" column="1">
       <widget class="QPushButton" name="removeShader">
        <property name="text">
         <string>Remove</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="0" rowspan="3">
       <widget class="QTextEdit" name="shader">
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="placeholderText">
         <string>No shader selected</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" alignment="Qt::AlignTop">
       <widget class="QPushButton" name="addShader">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>syncWithVideo</tabstop>
  <tabstop>syncToFramerate</tabstop>
  <tabstop>fpsSlider</tabstop>
  <tabstop>targetFps</tabstop>
  <tabstop>vsync</tabstop>
  <tabstop>shader</tabstop>
  <tabstop>addShader</tabstop>
  <tabstop>removeShader</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>OpenGLOptionsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>310</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>OpenGLOptionsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>310</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>syncToFramerate</sender>
   <signal>toggled(bool)</signal>
   <receiver>targetFps</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>140</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>380</x>
     <y>98</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>syncToFramerate</sender>
   <signal>toggled(bool)</signal>
   <receiver>fpsSlider</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>168</x>
     <y>87</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fpsSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>targetFps</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>252</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>308</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetFps</sender>
   <signal>valueChanged(int)</signal>
   <receiver>fpsSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>364</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>134</x>
     <y>93</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeShader</sender>
   <signal>clicked()</signal>
   <receiver>shader</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>201</y>
    </hint>
    <hint type="destinationlabel">
     <x>235</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
